src/main/scala/org/holmesprocessing/totem/services/peinfo/peinfov2.py

```diff
@@ -0,0 +1,347 @@
+    data = list()
+    for module in pe.DIRECTORY_ENTRY_DELAY_IMPORT:
+        module_list = list()
+        data['Delay Imported symbols'].append(module_list)
+        module_list.append(module.struct.dump_dict())
+
+        for symbol in module.imports:
+            symbol_dict = dict()
+            if symbol.import_by_ordinal is True:
+                symbol_dict['DLL'] = module.dll
+                symbol_dict['Ordinal'] = symbol.ordinal
+            else:
+                symbol_dict['DLL'] = module.dll
+                symbol_dict['Name'] = symbol.name
+                symbol_dict['Hint'] = symbol.hint
+
+            if symbol.bound:
+                symbol_dict['Bound'] = symbol.bound
+            module_list.append(symbol_dict)
+
+    return data
+
+def PEInfoRun(obj):
+    data = {}
+    try:
+        pe = pefile.PE(obj)
+    except pefile.PEFormatError as e:
+        return e
+  
+    data["HEADERS"] = _headers(pe)
+    data["Sections"] = _sections(pe)
+
+    # data["DllCharacteristics"] = _dll_characteristics_flags(pe)
+    
+    if (hasattr(pe, 'OPTIONAL_HEADER') and hasattr(pe.OPTIONAL_HEADER, 'DATA_DIRECTORY') ):
+         data["directories"] = _dataDirectories(pe)
+
+    if hasattr(pe, 'VS_VERSIONINFO'):
+         data["VersionInfo"] = _version_Information(pe)
+
+    if hasattr(pe, 'DIRECTORY_ENTRY_EXPORT'):
+        data['Exported symbols'] = _export_directory(pe)
+
+    if hasattr(pe, 'DIRECTORY_ENTRY_IMPORT'):
+        data['Importedsymbols'] = _imported_symbols(pe)
+        
+
+    if hasattr(pe, 'DIRECTORY_ENTRY_BOUND_IMPORT'):
+                data['Bound imports'] = _directory_bound_imports(pe)
+
+    if hasattr(pe, 'DIRECTORY_ENTRY_DELAY_IMPORT'):
+                data['Delay Imported symbols'] = _delay_import_directory(pe)
+
+    if hasattr(pe, 'DIRECTORY_ENTRY_RESOURCE'):
+            data['Resource directory'] = _resources_data_entry(pe)
+            
+    if ( hasattr(pe, 'DIRECTORY_ENTRY_TLS') and pe.DIRECTORY_ENTRY_TLS and pe.DIRECTORY_ENTRY_TLS.struct ):
+            data['TLS'] = pe.DIRECTORY_ENTRY_TLS.struct.dump_dict()
+
+
+    if ( hasattr(pe, 'DIRECTORY_ENTRY_LOAD_CONFIG') and pe.DIRECTORY_ENTRY_LOAD_CONFIG and pe.DIRECTORY_ENTRY_LOAD_CONFIG.struct ):
+        data['LOAD_CONFIG'] = pe.DIRECTORY_ENTRY_LOAD_CONFIG.struct.dump_dict()
+
+
+    if hasattr(pe, 'DIRECTORY_ENTRY_DEBUG'):
+        data['Debug information'] = _debug_Directory(pe)
+
+
+    if hasattr(pe, 'DIRECTORY_ENTRY_BASERELOC'):
+        data['Base relocations'] = _relocations_directory(pe)
+
+    return data
+
+class PEInfoProcess(tornado.web.RequestHandler):
+    def get(self):
+        try:
+            filename = self.get_argument("obj", strip=False)
+            fullPath = os.path.join('/tmp', filename)
+            start_time = time.time()
+            data = PEInfoRun(fullPath)
+            self.write(data)
+            print("--- Done analysing Total time taken %s ms --- \n" % ((time.time() - start_time)*1000))
```

Comment:
I would comment this out for the pull request. You can also create a debug flag.